File: Data Structures & Algorithms/pascals-triangle/submission-0.java
class Solution {
    public List<List<Integer>> generate(int nr) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> row = new ArrayList<>();
        for(int i = 1; i <= nr; i++)
        {
            row = new ArrayList<>();
            row.add(1);
            if(i > 2)
            {
                
                for(int j = 1; j < i - 1; j++)
                {
                    // System.out.println(i + "   " + j);
                    row.add(ans.get(i - 2).get(j) + ans.get(i - 2).get(j - 1));
                } 
            }
            if(i > 1) row.add(1);
            ans.add(new ArrayList<>(row));
        }
        return ans;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& pr) {
        int ans = 0;
        for(int i = 1; i < pr.size(); i++) if(pr[i] > pr[i - 1]) ans+=(pr[i] - pr[i - 1]);
        return ans;
    }
};